src/api/svg.ts

```diff
@@ -112,29 +129,60 @@ const runnersToPage = (
     const textWidth = (text.length * fontSize) / 2; // We try to approx the width of the text
     const offset =
       anchor === 'middle' ? textWidth / 2 : anchor === 'end' ? textWidth : 0;
-    page.drawText(text, {
-      x: (element.svgAttributes.x || 0) - offset,
-      y: element.svgAttributes.y || 0,
-      font:
-        options.fonts && element.svgAttributes.fontFamily
-          ? options.fonts[element.svgAttributes.fontFamily]
-          : undefined,
-      size: fontSize,
-      color: element.svgAttributes.fill,
-      opacity: element.svgAttributes.fillOpacity,
-      rotate: element.svgAttributes.rotate,
-    });
+    const point = { x: (element.svgAttributes.x || 0) - offset, y:  element.svgAttributes.y || 0 }
+
+    if (isCoordinateInsideTheRect(point, svgRect)) {
```

Comment:
From what I understand, when the text is not inside the bounds, it won't show up, right? Then, let pdf-lib handle this and for now don't worry about it. If you really want to do sth here, that would be to cut the text and draw only the letters that are inside the bound.

Reply:
That's true for the pdf's boundings, but not for the viewBox's bounding.
I agree that this implementation is incomplete, but it's enough for now.